HACKING.md

```diff
@@ -318,65 +318,152 @@ the build frontend is [build](https://pypi.org/project/build/).
-14. When you are ready, click "Publish". (If you are not ready, click "Save as Draft".)
-15. If the release includes an `ops-scenario` package, then create a new tag in
-   the form `scenario-<major>.<minor>.<patch>`.
-
-This will trigger automatic builds for the Python packages and publish them to
-PyPI ([ops](https://pypi.org/project/ops/) and
-[ops-scenario](https://pypi.org/project/ops-scenario)) (authorisation is handled via a
-[Trusted Publisher](https://docs.pypi.org/trusted-publishers/) relationship).
+9. Save the release notes as a draft, and have someone else in the Charm-Tech
+   team proofread the release notes.
+10. If the release includes both `ops` and `ops-scenario` packages, then push a
+   new tag in the form `scenario-<major>.<minor>.<patch>`. This is done by
+   executing `git tag scenario-x.y.z`, then `git push upstream --tags` locally
+   (assuming you have configured `canonical/operator` as a remote named
+   `upstream`).
+11. When you are ready, click "Publish". GitHub will create the additional tag.
+
+Pushing the tags will trigger automatic builds for the Python packages and
+publish them to PyPI ([ops](https://pypi.org/project/ops/) and
+[ops-scenario](https://pypi.org/project/ops-scenario)) (authorisation is handled
+via a [Trusted Publisher](https://docs.pypi.org/trusted-publishers/) relationship).
 Note that it sometimes take a bit of time for the new releases to show up.
 
-To do a release that does not include an `ops-scenario` package, skip the
-step above where a `scenario-` tag is created. To do a release that does not
-include an `ops` package, create the `scenario-<major>.<minor>.<patch>` tag as
-part of the GitHub release process instead of the `<major>.<minor>.<patch>` one.
-
 See [.github/workflows/publish-ops.yaml](.github/workflows/publish-ops.yaml) and
 [.github/workflows/publish-ops-scenario.yaml](.github/workflows/publish-ops-scenario.yaml) for details.
 (Note that the versions in the YAML refer to versions of the GitHub actions, not the versions of the ops library.)
 
 You can troubleshoot errors on the [Actions Tab](https://github.com/canonical/operator/actions).
 
-16. Announce the release on [Discourse](https://discourse.charmhub.io/c/framework/42) and [Matrix](https://matrix.to/#/#charmhub-charmdev:ubuntu.com). Mention both the `ops` and `ops-scenario` releases in the same posts, but avoid the word "Scenario", preferring "unit testing API" or "state transition testing".
+12. Announce the release on [Discourse](https://discourse.charmhub.io/c/framework/42) and [Matrix](https://matrix.to/#/#charmhub-charmdev:ubuntu.com).
+
+13. Open a PR to change the version strings to the expected
+   next version, with ".dev0" appended (for example, if 3.14.1 is the next
+   expected version, use `'3.14.1.dev0'`).
+
+## Release Documentation
+
+`ops` and `ops-scenario` releases have several documentation artifacts, each
+serving a separate purpose and covering a different level.
```

Comment:
This sentence left me wondering what these different artefacts are specifically and what the implications are for someone doing a release.

Aren't all our docs always tracking main, rather than specific releases?

Are the release artifacts changing from just being zipped source code?

Oh also today I learned that artifacts is US English, and artefacts is British English.

Reply:
```suggestion
`ops` and `ops-scenario` releases have several documentation artefacts, each
serving a separate purpose and covering a different level.
```

Reply:
> This sentence left me wondering what these different artefacts are specifically and what the implications are for someone doing a release.

Hmm, I maybe need to use a different term then. They are the things outlined in the subsequent sections: raw logs, changelogs, release notes, release post.

> Aren't all our docs always tracking main, rather than specific releases?

Yes, except for these documents that are specifically about the release.

> Are the release artifacts changing from just being zipped source code?

The zipped and tar.gz repository dump (what GitHub calls "assets") stay the same: I don't think we can change those, and no-one should really be using them. The CI pipeline produces .tar.gz and wheels of the two packages (unchanged for `ops`, minor change for `ops-scenario` in that it previously was wheel-only) that are uploaded to PyPI and attestations for each of those (unchanged for `ops`, new for `ops-scenario`). We manually create the `CHANGES.md` files (unchanged for `ops`, new for `ops-scenario`), which are included in both the "assets" and PyPI artefacts. We also manually create the GitHub Releases text and Discourse post, which only live in those places.

Any suggestions for a better/clearer title here? Or clearer text?

Reply:
Oh I see now. I just immediately jumped to build artefacts. Probably just my unfamiliarity with terminology. Mentally substituting with "We produce several pieces of documentation for `ops` and `ops-scenario` releases, each serving ..." makes everything make sense for me.